semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,477 @@
+        z = np.nan_to_num(z)
+        ip = CloughTocher2DInterpolator((x, y), z, fill_value=0)
+        irap_z = ip(irap_x, irap_y)
+
+        for i in range(len(surf)):
+            surf[i] = irap_z[i]
+
+        return surf
+
+    def add_survey(self, name, date):
+        """The add_survey() method will register a survey at a specific date.
+
+        The name argument should be a unique string, this will later
+        be used when evaluating the elastic strain in the
+        overburden. The date should be python datetime.date()
+        instance, this date should be present as a report step in the
+        restart file - otherwise an exception will be raised.
+        """
+        restart_view = self._rst_file.restartView(sim_time=date)
+        self.subsidence.add_survey_PRESSURE(name, restart_view)
+
+        self._restart_views[name] = restart_view
+
+        return restart_view
+
+    @staticmethod
+    def _divide_negative_shift(ts, div_val=5.0):
+        for i in range(len(ts)):
+            if ts[i] < 0:
+                ts[i] /= div_val
+
+    def geertsma_ts_rporv(self, vintage_pairs):
+        """
+        Calculates TS without using velocity. Fast.
+        Velocity is only used to get the surface on the velocity grid.
+        Uses change in porevolume from Eclipse (RPORV in .UNRST) as input to
+        Geertsma model.
+
+        :param vintage_pairs:
+        """
+        return self._geertsma_ts_custom(
+            vintage_pairs, self.subsidence.eval_geertsma_rporv, "TS_RPORV"
+        )
+
+    def geertsma_ts_simple(self, vintage_pairs):
+        """
+        Calculates TS without using velocity. Fast.
+        Velocity is only used to get the surface on the velocity grid.
+
+        :param vintage_pairs:
+        """
+        return self._geertsma_ts_custom(
+            vintage_pairs, self.subsidence.evalGeertsma, "TS_SIMPLE"
+        )
+
+    def _geertsma_ts_custom(self, vintage_pairs, subsidence_func, method_name):
+        """
+        Calculates TS without using velocity. Fast.
+
+        :param vintage_pairs:
+        :param subsidence_func: specify subsidence method to be used
+        :param method_name: string represeting the subsudence func name
+        """
+
+        if len(vintage_pairs) < 1:
+            return 0, []
+
+        vintages = self._vintages_name_date(vintage_pairs)
+        surface = self._surface
+        points_to_calculate = self._get_non_nan_points()
+        surf_displacement = np.zeros((len(vintages.name), len(surface)))
+        ts_surfaces = []
+        for idx_vtg, (name_vtg, date_vtg) in enumerate(zip(vintages.name, vintages.date)):
+            try:
+                logging.info(
+                    "{:%x %X} {}: Calculating vintage {:%Y.%m.%d}".format(
+                        dt.now(), method_name, date_vtg
+                    )
+                )
+            except TypeError:
+                logging.exception("Wrong date vintage format!")
```

Comment:
This message should be improved to help debugging, at least it should include the previous error to show where there is an error.

Reply:
Or - couldnt we validate the vintage data format earlier, to avoid this problem entirely?

Reply:
There's a test, which injects wrong (or None) dates into ots calculations directly:
https://github.com/xjules/semeio/blob/b16b75a8fd8905bac2fe5157657a6bb3c5b1e011/tests/jobs/ots/test_ots.py#L189

It expects that `ValueError` is risen, but since this won't be used through API (or at least that's what I reckon) that test can go away. What do you think?

Reply:
Yes, that seems reasonable, as long as the dates are validated that should be no problem.

Reply:
Yes, mainly if they are present in `ecl.UNRST` file

Reply:
This exception is gone as well as the (now unnecessary) test